docs/governance/rfc/005-python-implementation.md

```diff
@@ -0,0 +1,146 @@
+2. *No full-duplex streams*: http/2 transport is hard to support on both client and server; half-duplex is fine though
+
+### Justification for not supporting gRPC on the server
+
+gRPC is a poor match to Python's most commonly used asynchronous server framework, ASGI.
+A correct gRPC implementation requires control over very low-level details of the connection state and http/2 transport.
+These details are not exposed by ASGI, so supporting them would prevent this library's servers from being ASGI applications.
+
+That's an unacceptable trade-off to make because ASGI interoperability is important for integrating connect servers into larger applications, like those running on Django, Starlette or FastAPI.
+
+### Justification for not supporting gRPC on the client
+
+Implementing a correct gRPC client again requires low-level control over the transport stack.
+The connect client would need to accept an extremely low-level and generic transport, even when contacting connect servers, just to support gRPC.
+
+This is more acceptable than the server case, but still unpleasant.
+Users will expect that they can pass an httpx.Client or similar to configure instrumentation, authentication, and so on.
+If we use a low-level library like hyper-h2 to implement the client, we'd likely need to build a bespoke abstraction supporting features like authentication which would be quite a bit more scope to cover.
+
+### Justification for not supporting grpc-web
+
+The grpc-web project lacks a standalone specification.
+It is defined [only in relation to gRPC](https://github.com/grpc/grpc/blob/6c7e2a94f99747950397258a8de005a3d90210a1/doc/PROTOCOL-WEB.md).
+
+As a result, most Connect implementations of grpc-web do so by treating it as a special case of gRPC.
+But without a gRPC implementation to base off of, a Python Connect library would need to implement grpc-web "from scratch" which is much more complex.
+
+The benefits of doing this are outweighed by its complexity for now.
+grpc-web users choose it because it is compatible with browsers, but so is Connect.
+grpc-web could be a useful later addition to the features provided by a Python implementation, but it need not be in the initial implementation.
+
+### Justification for only supporting half-duplex streams
+
+This is related to the previous justifications on gRPC: http/2 is poorly supported in Python today.
+
+## Goals
+
+Fully pass conformance tests, with the following configuration:
+```
+features:
+  versions:
+    - HTTP_VERSION_1
+  protocols:
+    - PROTOCOL_CONNECT
+  codecs:
+    - CODEC_PROTO
+    - CODEC_JSON
+  compressions:
+    - COMPRESSION_IDENTITY
+    - COMPRESSION_GZIP
+    - COMPRESSION_BR
+    - COMPRESSION_ZSTD
+  stream_types:
+    - STREAM_TYPE_UNARY
+    - STREAM_TYPE_CLIENT_STREAM
+    - STREAM_TYPE_SERVER_STREAM
+    - STREAM_TYPE_HALF_DUPLEX_BIDI_STREAM
+  supports_half_duplex_bidi_over_http1: true
```

Comment:
I guess this answers the question above, that the intention is to support bidi streams, but only with half-duplex capabilities.

FWIW, we've considered this in [connect-go](https://github.com/connectrpc/connect-go/pull/796) as well, but we haven't actually enabled it because of potential UX issues for failure modes. Would it be acceptable to initially limit the scope to exclude bidi streams unless/until there's a good way to provide good error messages on misuse?